🐛 Building android with React Native 0.71

### What were you trying to do?

Hello,

I am trying to build my application for android.
On iOS, it builds well and launches.

I use `react-native-reanimated` : `2.14.1`.

I get these warings and errors on the android build : 
```bash
> Configure project :react-native-vision-camera
VisionCamera: node_modules/ found at: /Users/***/node_modules
VisionCamera: Building with JSC...
Warning: The 'kotlin-android-extensions' Gradle plugin is deprecated. Please use this migration guide (https://goo.gle/kotlin-android-extensions-deprecation) to start working with View Binding (https://developer.android.com/topic/libraries/view-binding) and the 'kotlin-parcelize' plugin.
VisionCamera: Frame Processors are enabled! Building C++ part...
WARNING:DSL element 'dexOptions' is obsolete and should be removed.
It will be removed in version 8.0 of the Android Gradle plugin.
Using it has no effect, and the AndroidGradle plugin optimizes dexing automatically.
```
```bash
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/***/node_modules/react-native-vision-camera/android/build.gradle' line: 279

* What went wrong:
A problem occurred evaluating project ':react-native-vision-camera'.
> Expected directory '/Users/***/node_modules/react-native/android' to contain exactly one file, however, it contains no files.
```

Any idea if vision-camera is ready for react-native 0.71?

Thank you for your work! 

### Reproduceable Code

_No response_

### What happened instead?

I get a gradle build error.

### Relevant log output

_No response_

### Device

Any

### VisionCamera Version

^2.15.2

### Additional information

- [ ] I am using Expo
- [X] I have read the [Troubleshooting Guide](https://mrousavy.github.io/react-native-vision-camera/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

> Ok I think I got it. Here's how you do library patching wrong, step by step:
> 
> 1. Since RN0.71 remove the local maven repos, we need to download them from maven and place them back to where they were. Namely, https://mvnrepository.com/artifact/com.facebook.react/react-android and  https://mvnrepository.com/artifact/com.facebook.react/hermes-android. For vision-camera you only need the aars (both debug and release varient), so go grab them.
> 2. Reconstruct the local mvn repo (or least reconstruct the parts we need) by creating the following two paths within your project's node_modules
>    `node_modules/react-native/android/com/facebook/react/react-native/{RN_FULL_VERSION}/`
>    `node_modules/react-native/android/com/facebook/react/hermes-engine/{RN_FULL_VERSION}/`
>    Replace RN_FULL_VERSION with the version you're using, say 0.71.0
> 3. Move the react-android aars into `node_modules/react-native/android/com/facebook/react/react-native/{RN_FULL_VERSION}/`, then replace the `react-android` filename prefix with `react-native`.
> 4. Move the hermes-android aars into `node_modules/react-native/android/com/facebook/react/hermes-engine/{RN_FULL_VERSION}/`, then replace the `hermes-android` filename prefix with `hermes-engine`.
> 5. I would recommend you build a script or something to do step 1 through 4
> 6. Next we patch your project. You'll need to modify several files. We'll start with top-level build.gradle - set `kotlinVersion =  "1.6.20"` in top-level build.gradle
> 7. We also need to modify app/build.gradle - add the following lines wherever you please
>    `project.ext.react = [   enableHermes: true, ]`
> 8. Finally we start modify `react-native-vision-camera`, navigate to the `react-native-vision-camera` folder within your node_modules and open `CMakeLists.txt`
> 9. Add the following line in `target_include_directories` - `"${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/jni/react/turbomodule"`
> 
> Now you should be able to use `react-native-vision-camera` as before.
> 
> I'll say this again: The above steps are how you do library patching wrong. It is "With Enough Thrust Pigs Fly Just Fine" in action and you should definitely do library patching in the other direction - patch the library, not the framework, and probably never patch the framework to make it closer to an older version.
> 
> I made this post just to highlight where things have changed and maybe provide some insights about how one could patch `react-native-vision-camera` in the right way to support RN0.71. So I won't disclose a repo or anything that automates the steps above.
> 
> Fun fact: I was never a fan of Java/Gradle/Maven or anything that's Java-bounded so there is zero chance I'll make a proper fix for this issue.
> 
> You guys have a nice day.

Thanks a lot of! This solution is great! I only have one problem in compilation:
`  ../../../../build/third-party-ndk/folly/folly/dynamic.h:63:10: fatal error: 'boost/operators.hpp' file not found`
 My stack is RN 0.71.1 and vision 2.15.2
 
 Do you know what can be happend?

--- a/android/CMakeLists.txt
+++ b/android/CMakeLists.txt
@@ -1,13 +1,26 @@
+project(VisionCamera)
 cmake_minimum_required(VERSION 3.4.1)
 
 set (CMAKE_VERBOSE_MAKEFILE ON)
 set (CMAKE_CXX_STANDARD 14)
-set (CMAKE_CXX_FLAGS "-DFOLLY_NO_CONFIG=1 -DFOLLY_HAVE_CLOCK_GETTIME=1 -DFOLLY_HAVE_MEMRCHR=1 -DFOLLY_USE_LIBCPP=1 -DFOLLY_MOBILE=1 -DON_ANDROID -DONANDROID -DFOR_HERMES=${FOR_HERMES}")
+
+if(${REACT_NATIVE_VERSION} GREATER_EQUAL 71)
+        include("${NODE_MODULES_DIR}/react-native/ReactAndroid/cmake-utils/folly-flags.cmake")
+        add_compile_options(${folly_FLAGS})
+else()
+        set (CMAKE_CXX_FLAGS "-DFOLLY_NO_CONFIG=1 -DFOLLY_HAVE_CLOCK_GETTIME=1 -DFOLLY_HAVE_MEMRCHR=1 -DFOLLY_USE_LIBCPP=1 -DFOLLY_MOBILE=1 -DON_ANDROID -DONANDROID -DFOR_HERMES=${FOR_HERMES}")
+endif()
+
 
 set (PACKAGE_NAME "VisionCamera")
 set (BUILD_DIR ${CMAKE_SOURCE_DIR}/build)
-set (RN_SO_DIR ${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/jni/first-party/react/jni)
-
+if(${REACT_NATIVE_VERSION} GREATER_EQUAL 71)
+        # Consume shared libraries and headers from prefabs
+        find_package(fbjni REQUIRED CONFIG)
+        find_package(ReactAndroid REQUIRED CONFIG)
+else()
+        set (RN_SO_DIR ${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/jni/first-party/react/jni)
+endif()
 # VisionCamera shared
 
 if(${REACT_NATIVE_VERSION} LESS 66)
@@ -36,64 +49,84 @@ add_library(
 )
 
 # includes
+if(${REACT_NATIVE_VERSION} GREATER_EQUAL 71)
+        target_include_directories(
+                ${PACKAGE_NAME}
+                PRIVATE
+                "${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/jni/react/turbomodule"
+                "${NODE_MODULES_DIR}/react-native/ReactCommon"
+                "${NODE_MODULES_DIR}/react-native/ReactCommon/callinvoker"
+                "${NODE_MODULES_DIR}/react-native/ReactCommon/jsi"
+                "${NODE_MODULES_DIR}/react-native/ReactCommon/react/renderer/graphics/platform/cxx"
+                "${NODE_MODULES_DIR}/react-native/ReactCommon/runtimeexecutor"
+                "${NODE_MODULES_DIR}/react-native/ReactCommon/yoga"
+                # --- Reanimated ---
+                # New
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/AnimatedSensor"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/Tools"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/SpecTools"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/SharedItems"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/Registries"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/LayoutAnimations"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/hidden_headers"
+                "src/main/cpp"
+        )
+else()
+        file (GLOB LIBFBJNI_INCLUDE_DIR "${BUILD_DIR}/fbjni-*-headers.jar/")
 
-file (GLOB LIBFBJNI_INCLUDE_DIR "${BUILD_DIR}/fbjni-*-headers.jar/")
+        target_include_directories(
+                ${PACKAGE_NAME}
+                PRIVATE
+                # --- fbjni ---
+                "${LIBFBJNI_INCLUDE_DIR}"
+                # --- Third Party (required by RN) ---
+                "${BUILD_DIR}/third-party-ndk/boost"
+                "${BUILD_DIR}/third-party-ndk/double-conversion"
+                "${BUILD_DIR}/third-party-ndk/folly"
+                "${BUILD_DIR}/third-party-ndk/glog"
+                # --- React Native ---
+                "${NODE_MODULES_DIR}/react-native/React"
+                "${NODE_MODULES_DIR}/react-native/React/Base"
+                "${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/jni"
+                "${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/java/com/facebook/react/turbomodule/core/jni"
+                "${NODE_MODULES_DIR}/react-native/ReactCommon"
+                "${NODE_MODULES_DIR}/react-native/ReactCommon/callinvoker"
+                "${NODE_MODULES_DIR}/react-native/ReactCommon/jsi"
+                "${NODE_MODULES_DIR}/hermes-engine/android/include/"
+                ${INCLUDE_JSI_CPP} # only on older RN versions
+                ${INCLUDE_JSIDYNAMIC_CPP} # only on older RN versions
+                # --- Reanimated ---
+                # Old
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/AnimatedSensor"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/Tools"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/SpecTools"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/SharedItems"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/Registries"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/LayoutAnimations"
+                # New
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/AnimatedSensor"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/Tools"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/SpecTools"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/SharedItems"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/Registries"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/LayoutAnimations"
+                "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/hidden_headers"
+                "src/main/cpp"
+        )
+endif()
 
-target_include_directories(
-        ${PACKAGE_NAME}
-        PRIVATE
-        # --- fbjni ---
-        "${LIBFBJNI_INCLUDE_DIR}"
-        # --- Third Party (required by RN) ---
-        "${BUILD_DIR}/third-party-ndk/boost"
-        "${BUILD_DIR}/third-party-ndk/double-conversion"
-        "${BUILD_DIR}/third-party-ndk/folly"
-        "${BUILD_DIR}/third-party-ndk/glog"
-        # --- React Native ---
-        "${NODE_MODULES_DIR}/react-native/React"
-        "${NODE_MODULES_DIR}/react-native/React/Base"
-        "${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/jni"
-        "${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/java/com/facebook/react/turbomodule/core/jni"
-        "${NODE_MODULES_DIR}/react-native/ReactCommon"
-        "${NODE_MODULES_DIR}/react-native/ReactCommon/callinvoker"
-        "${NODE_MODULES_DIR}/react-native/ReactCommon/jsi"
-        "${NODE_MODULES_DIR}/hermes-engine/android/include/"
-        ${INCLUDE_JSI_CPP} # only on older RN versions
-        ${INCLUDE_JSIDYNAMIC_CPP} # only on older RN versions
-        # --- Reanimated ---
-        # Old
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/AnimatedSensor"
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/Tools"
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/SpecTools"
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/SharedItems"
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/Registries"
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/headers/LayoutAnimations"
-        # New
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/AnimatedSensor"
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/Tools"
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/SpecTools"
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/SharedItems"
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/Registries"
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/LayoutAnimations"
-        "${NODE_MODULES_DIR}/react-native-reanimated/Common/cpp/hidden_headers"
-        "src/main/cpp"
-)
+
 
 # find libraries
 
 file (GLOB LIBRN_DIR "${BUILD_DIR}/react-native-0*/jni/${ANDROID_ABI}")
 
 if(${FOR_HERMES})
-        string(APPEND CMAKE_CXX_FLAGS " -DJS_RUNTIME_HERMES=1")
+        string(APPEND CMAKE_CXX_FLAGS " -DFOR_HERMES=1")
 
-        if(${REACT_NATIVE_VERSION} LESS 69)
-                # From `hermes-engine` npm package
-                target_include_directories(
-                        ${PACKAGE_NAME}
-                        PRIVATE
-                        "${JS_RUNTIME_DIR}/android/include"
-                )
-        else()
+        if(${REACT_NATIVE_VERSION} GREATER_EQUAL 71)
+                find_package(hermes-engine REQUIRED CONFIG)
+        elseif(${REACT_NATIVE_VERSION} GREATER_EQUAL 69)
                 # Bundled Hermes from module `com.facebook.react:hermes-engine` or project `:ReactAndroid:hermes-engine`
                 target_include_directories(
                         ${PACKAGE_NAME}
@@ -101,23 +134,41 @@ if(${FOR_HERMES})
                         "${JS_RUNTIME_DIR}/API"
                         "${JS_RUNTIME_DIR}/public"
                 )
+        else()
+                # From `hermes-engine` npm package
+                target_include_directories(
+                        ${PACKAGE_NAME}
+                        PRIVATE
+                        "${JS_RUNTIME_DIR}/android/include"
+                )
         endif()
 
-        target_link_libraries(
-                ${PACKAGE_NAME}
-                "${BUILD_DIR}/third-party-ndk/hermes/jni/${ANDROID_ABI}/libhermes.so"
-        )
+        if(${REACT_NATIVE_VERSION} GREATER_EQUAL 71)
+                target_link_libraries(
+                        ${PACKAGE_NAME}
+                        "hermes-engine::libhermes"
+                )
+        else()
+                target_link_libraries(
+                        ${PACKAGE_NAME}
+                        "${BUILD_DIR}/third-party-ndk/hermes/jni/${ANDROID_ABI}/libhermes.so"
+                )
+        endif()
         file (GLOB LIBREANIMATED_DIR "${BUILD_DIR}/react-native-reanimated-*-hermes.aar/jni/${ANDROID_ABI}")
 else()
         file (GLOB LIBJSC_DIR "${BUILD_DIR}/android-jsc*.aar/jni/${ANDROID_ABI}")
 
-        # Use JSC
-        find_library(
-                JS_ENGINE_LIB
-                jscexecutor
-                PATHS ${LIBRN_DIR}
-                NO_CMAKE_FIND_ROOT_PATH
-        )
+        if(${REACT_NATIVE_VERSION} GREATER_EQUAL 71)
+                set(JS_ENGINE_LIB ReactAndroid::jscexecutor)
+        else()
+                # Use JSC
+                find_library(
+                        JS_ENGINE_LIB
+                        jscexecutor
+                        PATHS ${LIBRN_DIR}
+                        NO_CMAKE_FIND_ROOT_PATH
+                )
+        endif()
         target_link_libraries(
                 ${PACKAGE_NAME}
                 ${JS_ENGINE_LIB}
@@ -127,12 +178,14 @@ else()
         file (GLOB LIBREANIMATED_DIR "${BUILD_DIR}/react-native-reanimated-*-jsc.aar/jni/${ANDROID_ABI}")
 endif()
 
-find_library(
-        FBJNI_LIB
-        fbjni
-        PATHS ${LIBRN_DIR}
-        NO_CMAKE_FIND_ROOT_PATH
-)
+if(${REACT_NATIVE_VERSION} LESS 71)
+        find_library(
+                FBJNI_LIB
+                fbjni
+                PATHS ${LIBRN_DIR}
+                NO_CMAKE_FIND_ROOT_PATH
+        )
+endif()
 
 if(${REACT_NATIVE_VERSION} LESS 69)
         find_library(
@@ -141,7 +194,7 @@ if(${REACT_NATIVE_VERSION} LESS 69)
                 PATHS ${LIBRN_DIR}
                 NO_CMAKE_FIND_ROOT_PATH
         )
-else()
+elseif(${REACT_NATIVE_VERSION} LESS 71)
         find_library(
                 FOLLY_LIB
                 folly_runtime
@@ -150,14 +203,25 @@ else()
         )
 endif()
 
-find_library(
-        REACT_NATIVE_JNI_LIB
-        reactnativejni
-        PATHS ${LIBRN_DIR}
-        NO_CMAKE_FIND_ROOT_PATH
-)
+if(${REACT_NATIVE_VERSION} LESS 71)
+        find_library(
+                REACT_NATIVE_JNI_LIB
+                reactnativejni
+                PATHS ${LIBRN_DIR}
+                NO_CMAKE_FIND_ROOT_PATH
+        )
+endif()
 
-if(${REACT_NATIVE_VERSION} LESS 66)
+if(${REACT_NATIVE_VERSION} GREATER_EQUAL 71)
+        target_link_libraries(
+                ${PACKAGE_NAME}
+                ReactAndroid::folly_runtime
+                ReactAndroid::glog
+                ReactAndroid::jsi
+                ReactAndroid::reactnativejni
+                fbjni::fbjni
+        )
+elseif(${REACT_NATIVE_VERSION} LESS 66)
         # JSI lib didn't exist on RN 0.65 and before. Simply omit it.
         set (JSI_LIB "")
 else()

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -50,7 +50,7 @@ def REACT_NATIVE_VERSION = reactProperties.getProperty("VERSION_NAME").split("\\
 def FOR_HERMES = System.getenv("FOR_HERMES") == "True"
 rootProject.getSubprojects().forEach({project ->
   if (project.plugins.hasPlugin("com.android.application")) {
-    FOR_HERMES = project.ext.react.enableHermes
+    FOR_HERMES = REACT_NATIVE_VERSION >= 71 && project.hermesEnabled || project.ext.react.enableHermes
   }
 })
 def jsRuntimeDir = {
@@ -130,6 +130,12 @@ android {
   buildToolsVersion getExtOrDefault('buildToolsVersion')
   ndkVersion getExtOrDefault('ndkVersion')
 
+  if (REACT_NATIVE_VERSION >= 71) {
+    buildFeatures {
+      prefab true
+    }
+  }
+
   defaultConfig {
     minSdkVersion 21
     targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
@@ -163,7 +169,17 @@ android {
 
   packagingOptions {
     // Exclude all Libraries that are already present in the user's app (through React Native or by him installing REA)
-    excludes = ["**/libc++_shared.so", "**/libfbjni.so", "**/libjsi.so", "**/libreactnativejni.so", "**/libfolly_json.so", "**/libreanimated.so", "**/libjscexecutor.so", "**/libhermes.so", "**/libfolly_runtime.so"]
+    excludes = ["**/libc++_shared.so",
+                "**/libfbjni.so",
+                "**/libjsi.so",
+                "**/libreactnativejni.so",
+                "**/libfolly_json.so",
+                "**/libreanimated.so",
+                "**/libjscexecutor.so",
+                "**/libhermes.so",
+                "**/libfolly_runtime.so",
+                "**/libglog.so",
+    ]
     // META-INF is duplicate by CameraX.
     exclude "META-INF/**"
   }
@@ -261,26 +277,36 @@ repositories {
 def kotlin_version = getExtOrDefault('kotlinVersion')
 
 dependencies {
-  // noinspection GradleDynamicVersion
-  implementation 'com.facebook.react:react-native:+'
+  if (REACT_NATIVE_VERSION >= 71) {
+      implementation "com.facebook.react:react-android:"
+      implementation "com.facebook.react:hermes-android:"
+  } else {
+      // noinspection GradleDynamicVersion
+      implementation 'com.facebook.react:react-native:+'
+  }
 
   if (ENABLE_FRAME_PROCESSORS) {
     implementation project(':react-native-reanimated')
 
-    //noinspection GradleDynamicVersion
-    extractHeaders("com.facebook.fbjni:fbjni:+:headers")
-    //noinspection GradleDynamicVersion
-    extractJNI("com.facebook.fbjni:fbjni:+")
+    if (REACT_NATIVE_VERSION < 71) {
+      //noinspection GradleDynamicVersion
+      extractHeaders("com.facebook.fbjni:fbjni:headers:+")
+      //noinspection GradleDynamicVersion
+      extractJNI("com.facebook.fbjni:fbjni:+")
 
-    def rnAarMatcher = "**/react-native/**/*${resolveBuildType()}.aar"
-    if (REACT_NATIVE_VERSION < 69) {
-      rnAarMatcher = "**/**/*.aar"
+      def rnAarMatcher = "**/react-native/**/*${resolveBuildType()}.aar"
+      if (REACT_NATIVE_VERSION < 69) {
+        rnAarMatcher = "**/**/*.aar"
+      }
+
+      def rnAAR = fileTree("$reactNative/android").matching({ it.include rnAarMatcher }).singleFile
+      def jscAAR = fileTree("${nodeModules}/jsc-android/dist/org/webkit/android-jsc").matching({ it.include "**/**/*.aar" }).singleFile
+      extractJNI(files(rnAAR, jscAAR))
     }
-    def rnAAR = fileTree("$reactNative/android").matching({ it.include rnAarMatcher }).singleFile
-    def jscAAR = fileTree("${nodeModules}/jsc-android/dist/org/webkit/android-jsc").matching({ it.include "**/**/*.aar" }).singleFile
+
     def jsEngine = FOR_HERMES ? "hermes" : "jsc"
     def reaAAR = "${nodeModules}/react-native-reanimated/android/react-native-reanimated-${REACT_NATIVE_VERSION}-${jsEngine}.aar"
-    extractJNI(files(rnAAR, jscAAR, reaAAR))
+    extractJNI(files(reaAAR))
   }
 
   implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
@@ -600,11 +626,13 @@ if (ENABLE_FRAME_PROCESSORS) {
 
   tasks.whenTaskAdded { task ->
     if (!task.name.contains('Clean') && (task.name.contains('externalNative') || task.name.contains('CMake'))) {
-      task.dependsOn(extractAARHeaders)
       task.dependsOn(extractJNIFiles)
-      task.dependsOn(prepareJSC)
-      task.dependsOn(prepareHermes)
-      task.dependsOn(prepareThirdPartyNdkHeaders)
+      if (REACT_NATIVE_VERSION < 71) {
+        task.dependsOn(extractAARHeaders)
+        task.dependsOn(prepareThirdPartyNdkHeaders)
+        task.dependsOn(prepareJSC)
+        task.dependsOn(prepareHermes)
+      }
     }
   }
 }

